src/openms/source/ANALYSIS/DECHARGING/MetaboliteFeatureDeconvolution.cpp

```diff
@@ -868,9 +868,9 @@ namespace OpenMS
         //remove info not wanted in pair
         std::vector<String> keys;
         cf.getKeys(keys);
-        for (std::vector<String>::const_iterator it = keys.begin(); it != keys.end(); ++it)
+        for (const auto& key : keys)
```

Comment:
```suggestion
        for (const String& key : keys)
```

Reply:
for simple types it is usually advised to just type it out instead of using auto for better readability